test/base/src/commonMain/kotlin/io/realm/test/util/Utils.kt

```diff
@@ -49,3 +49,12 @@ suspend fun <T : RealmObject> T.update(block: T.() -> Unit): T {
         liveObject
     }
 }
+
+// Expose a try-with-resource pattern for Realms
+fun Realm.use(action: (Realm) -> Unit) {
```

Comment:
Should we consider providing this for the user? I guess users don't need it that often that it is worth adding it.

Reply:
I would probably keep this out of the public API for now since we are pushing the idea that you don't really need to close a Realm again, so this method mostly seems useful for tests IMO.